internal/controller/config/config.go

```diff
@@ -33,36 +33,10 @@ func GetConfigManagerInstance() *ConfigManager {
 // supposed to be populated from the configmap attached to the
 // controller manager.
 type GlobalConfig struct {
-	ClusterName     string `json:"clusterName"`
-	TimeIntervalSec uint   `json:"timeIntervalSec"`
+	ClusterName     string `json:"clusterName" mapstructure:"clusterName"`
+	TimeIntervalSec uint   `json:"timeIntervalSec" mapstructure:"timeIntervalSec"`
```

Comment:
apparently, the "json" tags were working, but supposedly viper uses "mapstructure" tags. Adding both won't hurt anything.